Added setupkind.sh script (#37058)
istio/istio: samples/kind-lb/README.md

> # Set up kubernetes cluster with a load balancer

Reviewer: ```suggestion
# Create a KinD cluster with an external load balancer
```

Author: Thanks and made changes accordingly

This bash script sets up a k8s cluster using kind and metallb.

> The following software are required on your machine to run the script:

Reviewer: ```suggestion
The following dependencies must first be installed before running this script:

```

Author: Thanks and made changes accordingly

  1. kubectl
  2. kind
  3. docker

Usage:

```bash
   ./setupkind.sh --cluster-name cluster1 --k8s-release 1.22.1 --ip-octet 255

Where:
    -n|--cluster-name  - name of the k8s cluster to be created, cluster1 will be used if not given
    -r|--k8s-release   - the release of the k8s to setup, latest available if not given
    -s|--ip-octet      - the 3rd octet for public ip addresses, 255 if not given, valid range: 0-255
    -h|--help          - print the usage of this script
```

> The `ip-octet` parameter controls the IP range when metallb gets configured for
> allocating public IP address for a load balancer resource. The entire IP segement
> is based on the docker network. `KinD` creates a docker network named `kind` when
> a k8s cluster is created. This docker network's subnet dictates the public IP
> address range for the metallb provided IP addresses. This is not important when
> create just one k8s cluster, using default will be sufficient. When creating multiple
> k8s clusters on a single machine, it will be important to provide a value which
> should be between 1 and 255. For example, to create two clusters, one can invoke
> the script two times with the following parameters:

Reviewer: ```suggestion
The `ip-octet` parameter controls the IP range that metallb will use for allocating
the public IP addresses of load balancer resources.

The public IPs are dictated by the docker network's subnet which `KinD` creates when it 
creates a k8s cluster. The 3rd octet is 255 by default, which is fine if you create just one
kind cluster on a single machine. However, when running multiple k8s clusters, it is
important to provide different values for each cluster.

For example, to create two clusters, run the script two times with the following parameters:
```

Author: took most of the suggestion and also added a bit more explaining about the public IP makeup.

```bash
  ./setupkind.sh --cluster-name cluster1 --ip-octet 255
  ./setupkind.sh --cluster-name cluster2 --ip-octet 245
```

> 
> When work with istio external control plane, you can use this script to easily
> create multiple k8s clusters with load balancer so that endpoints can be easily
> accessible from multiple k8s clusters.

Reviewer: ```suggestion
```

Author: removed as suggested.

# Create a KinD cluster with an external load balancer

This bash script sets up a k8s cluster using kind and metallb.

The following dependencies must be installed before running the script:
  1. kubectl
  2. kind
  3. docker

Usage:

```bash
   ./setupkind.sh --cluster-name cluster1 --k8s-release 1.22.1 --ip-octet 255

Where:
   -n|--cluster-name - name of the k8s cluster to be created, cluster1 will be used if not given
   -r|--k8s-release  - the release of the k8s to setup, latest available if not given
   -s|--ip-octet     - the 3rd octet for public ip addresses, 255 if not given, valid range: 100-255
   -h|--help         - print the usage of this script
```

> The `ip-octet` parameter controls the IP range when metallb will use for allocating

Reviewer: ```suggestion
The `ip-octet` parameter controls the IP range that metallb will use for allocating
```

Author: done as suggested.

the public IP addresses for load balancer resources.

The public IPs are dictated by the docker network's subnet which `KinD` creates

> when `KinD` creates a k8s cluster. This parameter is used as the 3rd octet for the

Author: Done as suggested

public IP v4 addresses when a load balancer is created. The default value is 255.

> The first two octets are determined by `KinD` created docker network, the 4th octet

Reviewer: ```suggestion
The first two octets are determined by the docker network created by `KinD`, the 4th octet
```

Author: Done as suggested

> is hard coded as 200-240. As you might have guessed, for each k8s cluster, one can

Reviewer: ```suggestion
is hard coded as 200-240. As you might have guessed, for each k8s cluster one can
```

Author: done as suggested

> create at most 40 public IP v4 addresses. This parameter is not required when you
> create just one cluster, however, when running multiple k8s clusters, it is important
> to proivde different values for each cluster to avoid overlapping addresses.
> 
> For example, to create two clusters, run the script two times with the following

Reviewer: ```suggestion
create at most 40 public IP v4 addresses.

The `ip-octet` parameter is not required when you
create just one cluster, however, when running multiple k8s clusters it is important
to provide different values for each cluster to avoid overlapping addresses.
For example, to create two clusters, run the script two times with the following
```

Author: done as suggested
